File: SYP.Web/Modules/ServerTypes/Products/ProductCategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductCategoryRow } from "./ProductCategoryRow";

export interface ProductCategoryColumns {
    Name: Column<ProductCategoryRow>;
    SortOrder: Column<ProductCategoryRow>;
    IsActive: Column<ProductCategoryRow>;
}

export class ProductCategoryColumns extends ColumnsBase<ProductCategoryRow> {
    static readonly columnsKey = 'Products.ProductCategory';
    static readonly Fields = fieldsProxy<ProductCategoryColumns>();
}

File: SYP.Web/Modules/Catalog/ProductCategory/ProductCategoryColumns.cs
namespace SYP.Catalog;

[ColumnsScript("Catalog.ProductCategory")]
[BasedOnRow(typeof(ProductCategoryRow), CheckNames = true)]
public class ProductCategoryColumns
{
    [EditLink, Width(300)]
    public string Name { get; set; }

    [Width(80)]
    public int SortOrder { get; set; }

    [Width(80)]
    public bool IsActive { get; set; }
}